font/src/ft/mod.rs

```diff
@@ -242,14 +239,55 @@ impl FreeTypeRasterizer {
         pattern.add_style(style);
         pattern.add_pixelsize(self.pixel_size);
 
-        let font = fc::font_match(fc::Config::get_current(), &mut pattern)
+        self.query_font(pattern, desc)
+    }
+
+    fn query_font(&mut self, mut pattern: fc::Pattern, desc: &FontDesc) -> Result<FontKey, Error> {
+        let config = fc::Config::get_current();
+        let fonts = fc::font_sort(&config, &mut pattern, true)
             .ok_or_else(|| Error::MissingFont(desc.to_owned()))?;
-        self.face_from_pattern(&font).and_then(|pattern| {
+
+        let mut font_iter = fonts.into_iter();
+
+        let base_font = font_iter.next().ok_or_else(|| Error::MissingFont(desc.to_owned()))?;
+        let font_path = base_font.file(0).ok_or_else(|| Error::MissingFont(desc.to_owned()))?;
+
+        // Reload already loaded faces and drop their fallback faces
+        let font_key = if let Some(font_key) = self.keys.remove(&font_path) {
+            let fallback_list = self.fallback_lists.remove(&font_key).unwrap_or_default();
+
+            for font in fallback_list {
+                let font_path = font.file(0).unwrap();
```

Comment:
This unwrap could be `if let Some()`, but I just I'd like to keep it like that for now, since I don't get it, how your pattern couldn't have a path after you query the thing. I'll come back to it later, since I don't really like to unwrap things.

Reply:
Makes sense, yeah. I could see how file(0).unwrap() might be safe, but file(1).unwrap() might not be or something like that.